File: backend/internal/constants/document.go
package constants

// DocumentType enumerates document kinds.
const (
	DocTypePS    = "ps"
	DocTypeRL    = "rl"
	DocTypeCV    = "cv"
	DocTypeEssay = "essay"
)

// ValidDocumentTypes returns all accepted document types.
func ValidDocumentTypes() []string {
	return []string{DocTypePS, DocTypeRL, DocTypeCV, DocTypeEssay}
}

// IsValidDocumentType reports whether a type is known.
func IsValidDocumentType(s string) bool {
	for _, v := range ValidDocumentTypes() {
		if v == s {
			return true
		}
	}
	return false
}

File: frontend/src/constants/document.ts
export type DocumentType = 'ps' | 'rl' | 'cv' | 'essay'

export const DocumentTypeMap: Record<DocumentType, string> = {
  ps: '个人陈述',
  rl: '推荐信',
  cv: '简历',
  essay: 'Essay',
}
